File: Contador de texto.js
document.addEventListener('DOMContentLoaded', () => {
    const textoEntrada = document.getElementById('textoEntrada');
    const botonAnalizar = document.getElementById('botonAnalizar');
    const conteoTotalCaracteresSinEspaciosElemento = document.getElementById('conteoTotalCaracteresSinEspacios');
    const conteoLetrasElemento = document.getElementById('conteoLetras');
    const conteoNumerosElemento = document.getElementById('conteoNumeros');
    const conteoCaracteresEspecialesElemento = document.getElementById('conteoCaracteresEspeciales');
    const conteoPalabrasCompletasElemento = document.getElementById('conteoPalabrasCompletas');
    const conteoElementosElemento = document.getElementById('conteoElementos');
    const conteoNumerosYCaracteresEspecialesElemento = document.getElementById('conteoNumerosYCaracteresEspeciales');
    const mensajePalabrasVaciasElemento = document.getElementById('mensajePalabrasVacias');
    const resultadosDiv = document.getElementById('resultados');
    
    botonAnalizar.addEventListener('click', () => {
        const texto = textoEntrada.value.trim();
        const elementos = texto.split(/\s+/).filter(item => item !== ''); // Consideramos palabras y números separados por espacio como elementos
        const soloPalabrasCompletas = elementos.filter(item => /^[a-zA-Z]+$/.test(item));
        const soloNumeros = texto.match(/\b\d+(\.\d+)?\b/g) || [];
        const caracteresEspeciales = texto.match(/[^\w\s\d]/g) || [];
        const totalCaracteresSinEspacios = texto.replace(/\s/g, '').length;
        const totalLetras = texto.replace(/[^a-zA-Z]/g, '').length;
        
        if (texto === '') {
            mensajePalabrasVaciasElemento.textContent = 'Por favor, ingrese algún texto.';
            mensajePalabrasVaciasElemento.style.display = 'block';
            if (resultadosDiv) {
                resultadosDiv.style.display = 'none';
            }
            return;
        }
        
        mensajePalabrasVaciasElemento.style.display = 'none';
        if (resultadosDiv) {
            resultadosDiv.style.display = 'block';
        }
        
        conteoTotalCaracteresSinEspaciosElemento.textContent = totalCaracteresSinEspacios;
        conteoLetrasElemento.textContent = totalLetras;
        conteoNumerosElemento.textContent = soloNumeros ? soloNumeros.length : 0;
        conteoCaracteresEspecialesElemento.textContent = caracteresEspeciales.length;
        conteoPalabrasCompletasElemento.textContent = soloPalabrasCompletas.length;
        conteoElementosElemento.textContent = elementos.length;
        conteoNumerosYCaracteresEspecialesElemento.textContent = (soloNumeros ? soloNumeros.length : 0) + caracteresEspeciales.length;
        
        if (soloPalabrasCompletas.length === 0) {
            mensajePalabrasVaciasElemento.style.display = 'block';
            mensajePalabrasVaciasElemento.textContent = 'No se encontraron palabras completas en el texto.';
        } else {
            mensajePalabrasVaciasElemento.style.display = 'none';
        }
    });
});

File: Contador de texto.py
import re

def analizar_texto_python_iterativo(texto_entrada):
    texto = texto_entrada.strip()
    elementos = [item for item in texto.split() if item]
    soloPalabrasCompletas = [item for item in elementos if re.fullmatch(r'[a-zA-Z]+', item)]
    soloNumeros = re.findall(r'\b\d+(?:\.\d+)?\b', texto)  # Corregido: el grupo no debe ser capturante si vas a contar

    total_digitos = 0
    total_caracteres_especiales = 0
    for caracter in texto:
        if caracter.isdigit():
            total_digitos += 1
        elif not caracter.isalnum() and not caracter.isspace():
            total_caracteres_especiales += 1

    totalCaracteresSinEspacios = len(re.sub(r'\s', '', texto))
    totalLetras = len(re.sub(r'[^a-zA-Z]', '', texto))
    totalElementos = len(elementos)
    totalNumerosYCaracteresEspeciales = len(soloNumeros) + total_caracteres_especiales

    mensajePalabrasVacias = ""
    if not soloPalabrasCompletas:
        mensajePalabrasVacias = "No se encontraron palabras completas en el texto."

    resultados = {
        "conteoTotalCaracteresSinEspaciosElemento": totalCaracteresSinEspacios,
        "conteoLetrasElemento": totalLetras,
        "conteoNumerosElemento": len(soloNumeros),
        "conteoCaracteresEspecialesElemento": total_caracteres_especiales,
        "conteoPalabrasCompletasElemento": len(soloPalabrasCompletas),
        "conteoElementosElemento": totalElementos,
        "conteoNumerosYCaracteresEspecialesElemento": totalNumerosYCaracteresEspeciales,
        "mensajePalabrasVaciasElemento": mensajePalabrasVacias
    }
    return resultados

if __name__ == "__main__":
    texto_ingresado = input("Ingrese el texto aquí: ")
    resultados_analisis = analizar_texto_python_iterativo(texto_ingresado)
    print("\nResultados del Análisis (con conteo iterativo):")
    print(f"Cantidad total de letras, números y caracteres: {resultados_analisis['conteoTotalCaracteresSinEspaciosElemento']}")
    print(f"Cantidad total de letras: {resultados_analisis['conteoLetrasElemento']}")
    print(f"Cantidad total de números: {resultados_analisis['conteoNumerosElemento']}")
    print(f"Cantidad total de caracteres especiales: {resultados_analisis['conteoCaracteresEspecialesElemento']}")
    print(f"Cantidad total de palabras completas: {resultados_analisis['conteoPalabrasCompletasElemento']}")
    print(f"Cantidad total de conjunto de elementos (palabras y números): {resultados_analisis['conteoElementosElemento']}")
    print(f"Cantidad total de números y caracteres especiales: {resultados_analisis['conteoNumerosYCaracteresEspecialesElemento']}")
    if resultados_analisis['mensajePalabrasVaciasElemento']:
        print(f"{resultados_analisis['mensajePalabrasVaciasElemento']}")